netket/_vmc.py

```diff
@@ -145,13 +146,20 @@ def _forward_and_backward(self):
             self._grads, self._jac = self._machine.vector_jacobian_prod(
                 samples_r, eloc_r / self._n_samples, self._grads, return_jacobian=True
             )
-            _sum_inplace(self._grads)
+            
+            self._grads = tree_map(_sum_inplace, self._grads)
```

Comment:
this breaks all other cases in which we are not using jax machines, is that correct?

Reply:
Yes, I will fix that and make sure other cases work with the updated code

Reply:
I was wrong, this doesn't things break things because the function tree_map returns _sum_inplace(self._grads) if self._grads is an array

Reply:
Indeed. `tree_map` operates on a jax-like pytree structure where the leaves are arrays (or jax DeviceArrays etc.). A single leave is a valid input, so `tree_map` should work with it, otherwise that'd be a bug in `tree_map`.